### PR TITLE
Fan et al. 2022

### LINKED ISSUE #93: Suggestion: Add `-y` when installing `pytest` in README
Just wanted to add a small suggestion for the README file. 

After the user creates their own Conda environment called, `microhapdb`, the user can copy-paste the following two lines.

>conda install pytest
pytest --pyargs microhapdb --doctest-modules

When a users copy-pastes these lines into the terminal, the terminal will execute the first line but will hang up on, "Does the user want to proceed downloading these packages.. (y/n): etc." and will not execute `pytest` until the users says yes. To streamline the execution, we could add the `-y` in the first command so that way it automatically says yes to the incoming packages and then immediately execute pytest when downloaded.

> conda install pytest -y

### LINKED ISSUE #95: Fan et al 2022
## Paper

Fan H, Xie Q, Wang L, Ru K, Tan X, Ding J, Wang X, Huang J, Wang Z, Li Y, Wang X, He Y, Gu C, Liu M, Ma S, Wen S, Qiu P (2022) Microhaplotype and Y-SNP/STR (MY): A novel MPS-based system for genotype pattern recognition in two-person DNA mixtures. *FSI: Genetics*, 59:102705, [doi:10.1016/j.fsigen.2022.102705](https://doi.org/10.1016/j.fsigen.2022.102705).

## Data URL(s)

Table S4

[Fan mmc2 see Suppl Tables 3 and 4.xlsx](https://github.com/bioforensics/MicroHapDB/files/9891525/Fan.mmc2.see.Suppl.Tables.3.and.4.xlsx)

[Fan mmc2 see Suppl Tables 3 and 4.txt](https://github.com/bioforensics/MicroHapDB/files/9975005/Fan.mmc2.see.Suppl.Tables.3.and.4.txt)

https://doi.org/10.1016/j.fsigen.2022.102705

## Author and PI information

Shaoqing Wen, wenshaoqing@fudan.edu.cn

Pingming Qiu, qiupmfy@126.com

### PR DESCRIPTION
**Pull request by standage**
This branch adds a new marker collection from [Fan et al. 2022](https://doi.org/10.1016/j.fsigen.2022.102705). All 22 of these markers include SNPs with no RSIDs, so population frequency estimates based on 1000 genomes haplotypes cannot be determined.

Closes #93. Closes #95.

### CODE REVIEW

File: dbbuild/sources/fan2022/Snakefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        bed38="marker38.bed",
+        chain="hg38ToHg19.over.chain.gz",
+    output:
+        bed37="marker37.bed",
+        bedunmapped="marker-unmapped.bed",
+    shell:
+        """
+        liftOver {input} {output}
+        [ ! -s {output.bedunmapped} ]
+        """
```
Comment (standage) [inline dbbuild/sources/fan2022/Snakefile:24]: Many of the SNPs/InDels don't have RSIDs, so positions for GRCh37 and GRCh38 must be provided explicitly. The authors provide only GRCh38 positions, so I whipped out UCSC's 38 --> 37 liftOver chain file for that conversion. Yay.

File: dbbuild/Snakefile [hunk 2]
```diff
@@ -374,4 +374,4 @@ rule indels:
             df = pandas.read_csv(freqfile, sep='\t', index_col=None)
             dfs.append(df)
         data = pandas.concat(dfs, axis=0, ignore_index=True)
-        data.sort_values('Marker').to_csv(output[0], sep='\t', index=False)
+        data.sort_values(['Marker', 'VariantIndex']).to_csv(output[0], sep='\t', index=False)
```
Comment (standage) [inline dbbuild/Snakefile:377]: Made a slight change here...

File: dbbuild/indels.tsv [hunk 3]
```diff
@@ -1,21 +1,46 @@
...
 mh05ZBF-001	1	TAAGTTAA	TAA
-mh06PK-24844	2	G	GA
+mh06FHL-001	2	C	CATT
+mh06FHL-001	7	A	AC
+mh06FHL-002	0	AT	A,GT
+mh06FHL-002	14	TG	T
+mh06FHL-002	25	C	CA
+mh06FHL-002	26	A	AG
 mh06PK-24844	1	C	CT
+mh06PK-24844	2	G	GA
```
Comment (standage) [inline dbbuild/indels.tsv:19]: ...to enforce a consistent order here.

**Comment by rnmitchell** [comment]
This looks like a Monday morning job... :)

**Comment by standage** [comment]
No rush!

**Comment by standage** [comment]
Ok this is ready for review @rnmitchell!

File: dbbuild/sources/fan2022/Snakefile (hunk 1 above)
Comment (rnmitchell) [reply dbbuild/sources/fan2022/Snakefile:24]: oh fun!!

**Comment by rnmitchell** [comment]
Looks good!

**Pull request merged**